### PR TITLE
Updated Maven sample

### PR DESCRIPTION
**Pull request by jglick**
@reviewbybees

### CODE REVIEW

File: src/main/js/samples.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ samples.push({
         "   stage 'Checkout'\n" +
         "\n" +
         "   // Get some code from a GitHub repository\n" +
-        "   git url: 'https://github.com/jglick/simple-maven-project-with-tests.git'\n" +
+        "   git 'https://github.com/jglick/simple-maven-project-with-tests.git'\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: just for fun

File: src/main/js/samples.js [hunk 2]
```diff
@@ -71,7 +71,12 @@ samples.push({
         "   // Mark the code build 'stage'....\n" +
         "   stage 'Build'\n" +
         "   // Run the maven build\n" +
-        "   sh \"${mvnHome}/bin/mvn -Dmaven.test.failure.ignore clean package\"\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: Only worked on Unix, and failed to account for the possibility of a workspace path with spaces.

File: src/main/js/samples.js [hunk 3]
```diff
@@ -71,7 +71,12 @@ samples.push({
...
-        "   step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])\n" +
+        "   if (isUnix()) {\n" +
+        "      sh \"'${mvnHome}/bin/mvn' -Dmaven.test.failure.ignore clean package\"\n" +
+        "   } else {\n" +
+        "      bat(/\"${mvnHome}\\bin\\mvn\" -Dmaven.test.failure.ignore clean package/)\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: Somewhat unusual Groovy quote syntax is better for Windows commands.

File: src/main/js/samples.js [hunk 4]
```diff
@@ -71,7 +71,12 @@ samples.push({
...
+        "   }\n" +
+        "   junit '**/target/surefire-reports/TEST-*.xml'\n" + // assumes junit & workflow-basic-steps up to date
```
Comment (jglick) [inline src/main/js/samples.js]: Less ugly!

File: src/main/js/samples.js [hunk 5]
```diff
@@ -71,7 +71,12 @@ samples.push({
...
+        "   archive 'target/*.jar'\n" + // TODO Jenkins 2 use archiveArtifacts instead
```
Comment (jglick) [inline src/main/js/samples.js]: And was just forgotten before.

**Comment by abayer** [comment]
:bee:

File: src/main/js/samples.js (hunk 2 above)
Comment (jglick) [inline src/main/js/samples.js]: [JENKINS-33707](https://issues.jenkins-ci.org/browse/JENKINS-33707)

**Pull request merged**